### PR TITLE
Halloween Forever

### PR DESCRIPTION
**Pull request by tabreturn**
# New Port for { Game Title }

## Game Information
- **Title**: Halloween Forever
- **URL**: https://imaginarymonsters.itch.io/halloween-forever

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [x] AmberELEC
- [x] ArkOS
- [x] ROCKNIX
- [x] muOS

### Resolution Tests
Test all major resolutions:
- [ ] 480x320
- [x] 640x480
- [x] 720x720 (RGB30)
- [x] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

File: ports/halloweenforever/Halloween Forever.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+source $controlfolder/control.txt
+# device_info.txt will be included by default
+
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+
+export PORT_32BIT="Y"
```
Comment (kloptops) [inline ports/halloweenforever/Halloween Forever.sh:21]: This should be before `mod_${CFW_NAME}.txt` 

```suggestion
export PORT_32BIT="Y"
```

File: ports/halloweenforever/Halloween Forever.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+elif [ -d "$XDG_DATA_HOME/PortMaster/" ]; then
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+source $controlfolder/control.txt
+# device_info.txt will be included by default
+
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
```
Comment (kloptops) [inline ports/halloweenforever/Halloween Forever.sh:18]:
```suggestion
export PORT_32BIT="Y"
[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
```

**Review by kloptops — CHANGES_REQUESTED** [review]

**Review by kloptops — DISMISSED** [review]

File: ports/halloweenforever/Halloween Forever.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+[ -f "./gamedata/game.win" ] && mv gamedata/game.win gamedata/game.droid
+[ -f "./gamedata/game.unx" ] && mv gamedata/game.unx gamedata/game.droid
+
+# Delete unneeded files
+rm -f gamedata/*.{dll,ini,exe}
+
+$GPTOKEYB "gmloader" -c ./halloweenforever.gptk &
+
+$ESUDO chmod +x "$GAMEDIR/gmloader"
+pm_platform_helper $GAMEDIR/gmloader
```
Comment (Cebion) [inline ports/halloweenforever/Halloween Forever.sh:46]:
```suggestion
pm_platform_helper "$GAMEDIR/gmloader"
```

File: ports/halloweenforever/port.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      "platformer",
+      "adventure"
+    ],
+    "image": null,
+    "rtr": false,
+    "exp": false,
+    "runtime": null,
+    "reqs": [],
+    "arch": [
+      "aarch64"
```
Comment (Cebion) [inline ports/halloweenforever/port.json:28]: Gmloader is 32 bit
```suggestion
      "armhf"
```

**Review by Cebion — CHANGES_REQUESTED** [review]

File: ports/halloweenforever/halloweenforever/halloweenforever.apk
Comment (cdeletre) [inline ports/halloweenforever/halloweenforever/halloweenforever.apk:1]: remove these files from the apk :)

```
unzip -l halloweenforever.apk
Archive:  halloweenforever.apk
  Length      Date    Time    Name
---------  ---------- -----   ----
     5472  00-00-1980 00:00   AndroidManifest.xml                <=====
     1361  00-00-1980 00:00   META-INF/CERT.RSA                  <=====
     1877  00-00-1980 00:00   META-INF/CERT.SF                   <=====
     1834  00-00-1980 00:00   META-INF/MANIFEST.MF               <=====
  1406608  00-00-1980 00:00   classes.dex                        <=====
  4772484  00-00-1980 00:00   lib/armeabi-v7a/libyoyo.so
  4969108  00-00-1980 00:00   lib/armeabi/libyoyo.so
  6850476  00-00-1980 00:00   lib/mips/libyoyo.so                <=====
  6055484  00-00-1980 00:00   lib/x86/libyoyo.so                 <=====
      244  00-00-1980 00:00   res/drawable-hdpi-v4/icon.png      <=====
      217  00-00-1980 00:00   res/drawable-ldpi-v4/icon.png      <=====
      228  00-00-1980 00:00   res/drawable-mdpi-v4/icon.png      <=====
      270  00-00-1980 00:00   res/drawable-xhdpi-v4/icon.png     <=====
      323  00-00-1980 00:00   res/drawable-xxhdpi-v4/icon.png    <=====
      392  00-00-1980 00:00   res/drawable-xxxhdpi-v4/icon.png.  <=====
      270  00-00-1980 00:00   res/drawable/icon.png              <=====
     5072  00-00-1980 00:00   res/layout/download.xml            <=====
      808  00-00-1980 00:00   res/layout/main.xml                <=====
     1528  00-00-1980 00:00   res/layout/userpasslayout.xml      <=====
     1040  00-00-1980 00:00   res/xml/preferences.xml            <=====
     5448  00-00-1980 00:00   resources.arsc                     <=====
---------                     -------
 24080544                     21 files
```

**Review by cdeletre — CHANGES_REQUESTED** [review]

File: ports/halloweenforever/halloweenforever/halloweenforever.apk
Comment (tabreturn) [reply ports/halloweenforever/halloweenforever/halloweenforever.apk:1]: Thanks. Done.

**Review by Cebion — APPROVED** [review]

**Review by kloptops — APPROVED** [review]

**Pull request merged**